lib/NetPacksLib.cpp

```diff
@@ -676,8 +676,24 @@ DLL_LINKAGE void GiveHero::applyGs(CGameState *gs)
 
 DLL_LINKAGE void NewObject::applyGs(CGameState *gs)
 {
-	const TerrainTile &t = gs->map->getTile(pos);
-	ETerrainType terrainType = t.terType;
+	ETerrainType terrainType;
+
+	if(ID == Obj::BOAT && !gs->isInTheMap(pos)) //special handling for bug #3060 - pos outside map but visitablePos is not
+	{
+		CGObjectInstance testObject = CGObjectInstance();

```

Comment:
Consider name testBoat.

Reply:
Or maybe testObject is fine here.

Reply:
Btw are you sure it is not an issue with position conversion to h3m format? I hardly understand what I am talking about.

Reply:
I am sure. The bug appears during boat purchase when shipyard is 1 tile away from map end, and there is water at the border of the map.